src/main.rs

```diff
@@ -401,36 +401,31 @@ fn run() -> Result<(), failure::Error> {
     if let Some(matches) = matches.subcommand_matches("config") {
         let api_key = matches.is_present("api-key");
 
-        let mut user = GlobalUser {
-            email: None,
-            api_key: None,
-            api_token: None,
-        };
+        let mut user: GlobalUser;
 
         if !api_key {
             // Default: use API token.
             message::info("Looking to use a Global API Key and email instead? Run \"wrangler config --api-key\". (Not Recommended)");
             println!("Enter API token: ");
             let mut api_token_str: String = read!("{}\n");
             api_token_str.truncate(api_token_str.trim_end().len());
-            if !api_token_str.is_empty() {
-                user.api_token = Some(api_token_str);
-            }
+            user = GlobalUser::TokenAuthUser {
+                api_token: api_token_str,
```

Comment:
nit: could change `api_token_str` to `api_token` and then you wouldn't need `api_token: api_token_str` you could just do `api_token`

Reply:
i actually don't see any existing `api_token` variable in scope, so the qualifier seems unnecessary